Lib/test/test_ctypes/test_dlerror.py

```diff
@@ -111,13 +110,90 @@ def test_null_dlsym(self):
             self.assertEqual(os.read(pipe_r, 2), b'OK')
 
             # Case #3: Test 'py_dl_sym' from Modules/_ctypes/callproc.c
-            L = dlopen(dstname)
+            L = _ctypes.dlopen(dstname)
             with self.assertRaisesRegex(OSError, "symbol 'foo' not found"):
-                dlsym(L, "foo")
+                _ctypes.dlsym(L, "foo")
 
             # Assert that the IFUNC was called
             self.assertEqual(os.read(pipe_r, 2), b'OK')
 
 
+@unittest.skipUnless(os.name != 'nt', 'test requires dlerror() calls')
+class TestLocalization(unittest.TestCase):
+
+    @staticmethod
+    def configure_locales(func):
+        return test.support.run_with_locale(
+            'LC_ALL',
+            'fr_FR.iso88591', 'ja_JP.sjis', 'zh_CN.gbk',
+            'fr_FR.utf8', 'en_US.utf8',
+            '',
+        )(func)
+
+    @classmethod
+    def setUpClass(cls):
+        if not has_gcc():
+            raise unittest.SkipTest("gcc is missing")
+
+    def make_empty_lib(self, outdir, so_libname):
+        srcname = os.path.join(outdir, 'empty.c')
+        dstname = os.path.join(outdir, so_libname)
+        create_empty_file(srcname)
+        args = ['gcc', '-shared', '-o', dstname, srcname]
+        p = subprocess.run(args, capture_output=True)
+        p.check_returncode()
+        return dstname
+
+    @configure_locales
+    def test_localized_error_from_dll(self):
+        with temp_dir() as outdir:
+            dstname = self.make_empty_lib(outdir, 'test_from_dll.so')
```

Comment:
Could not we use an existing library for these tests? They should not require gcc.

See for example test_loading.

Reply:
Yes, we can do it.